src/libcore/ffi.rs

```diff
@@ -49,8 +49,10 @@ impl fmt::Debug for c_void {
 /// Basic implementation of a `va_list`.
 // The name is WIP, using `VaListImpl` for now.
 #[cfg(any(all(not(target_arch = "aarch64"), not(target_arch = "powerpc"),
-              not(target_arch = "x86_64"), not(target_arch = "asmjs")),
+              not(target_arch = "x86_64")),
           all(target_arch = "aarch64", target_os = "ios"),
+          target_arch = "wasm32",
+          target_arch = "js",
```

Comment:
I'm not sure this is actually tested, but it's definitely more correct than the code I deleted below.